proxy/grpcproxy/client.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2016 The etcd Authors
```

Comment:
as it is now:
test <-> grpcproxy/kvclient <-> grpcproxy/kvproxy <-> clientv3/ <-> (socket <-> etcdserver)

is this the way you're proposing?
test <-> clientv3/kvclient <-> socket <-> (grpcproxy/kvproxy <-> clientv3/ <-> socket <-> etcdserver)

I think both are fine depending on what's being tested. The first one matches the proxy data model with the etcd server data model. The second one does that plus extra connection jiggling but I think that's better suited to e2e tests; what happens when the client loses contact with the proxy shouldn't be different from what happens when it loses contact with a regular etcd server since all the reconnection logic is in the client.

not particularly attached to file name, will put more thought into it